server/src/main/java/org/elasticsearch/search/fetch/subphase/InnerHitsContext.java

```diff
@@ -130,7 +133,17 @@ public SourceLookup getRootLookup() {
         }
 
         public void setRootLookup(SourceLookup rootLookup) {
-            this.rootLookup = rootLookup;
+            this.rootLookup.setSource(rootLookup.source());
+            this.rootLookup.setSourceContentType(rootLookup.sourceContentType());
+        }
+
+        @Override
+        public SearchLookup getSearchLookup() {
```

Comment:
Slightly tricksy; inner hits goes and runs a new FetchPhase for each document, and because positioning of the ValuesLookup is now done by the FetchPhase we need to have a new lookup for each inner hits context so that documents with multiple nested types don't interfere with each other.  We still only have one lookup per nested type throughout the phase so the overhead is kept low.